intelmq/lib/bot.py

```diff
@@ -752,16 +757,21 @@ def __load_runtime_configuration(self):
 
         if self.__bot_id in config:
             params = config[self.__bot_id]
-            self.run_mode = params.get('run_mode', 'continuous')
-            self.name = params.get('name')
-            self.description = params.get('description')
-            self.group = params.get('group')
-            self.module = params.get('module')
+            for option, value in params.items():
+                if option in ALLOWED_SYSTEM_PARAMETERS and value:
+                    self.__log_configuration_parameter("system", option, value)
+                    setattr(self, option, value)
+                elif option not in IGNORED_SYSTEM_PARAMETERS:
+                    self.logger.warning('Ignoring disallowed system parameter %r.',
+                                        option)
```

Comment:
Whats the difference between an `option` that is not in `ALLOWED_SYSTEM_PARAMETERS` and an option that is in `IGNORED_SYSTEM_PARAMETERS` (besides the latter is logged)? Why not simply drop `IGNORED_SYSTEM_PARAMETERS` and log all parameters that are not in `ALLOWED_SYSTEM_PARAMETERS`?
Nitpicking: I prefer iterating through dicts using `key, value` instead of `option, value`. If you don't want to use `key` I'd prefer the term `parameter` instead of `option` to make it more consistent.

Reply:
Ad naming: I just used the same naming as in the iteration below

Reply:
`IGNORED_SYSTEM_PARAMETERS` are silently ignored, as they are legit, but irrelevant here.
Everything which is not in `ALLOWED_SYSTEM_PARAMETERS` or `IGNORED_SYSTEM_PARAMETERS` is not allowed and likely a user configuration error (e.g. things that should have gone into `parameters`)

Reply:
Fixed the naming